ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/verkle/cache/TrieKeyPreloader.java

```diff
@@ -0,0 +1,258 @@
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class TrieKeyPreloader {
+  private static final int TRIE_KEY_CACHE_SIZE = 10_000;
+  private final Cache<Address, Map<Bytes32, Bytes32>> trieKeysCache =
+      CacheBuilder.newBuilder().maximumSize(TRIE_KEY_CACHE_SIZE).build();
+
+  private final TrieKeyAdapter trieKeyAdapter;
+
+  private final Hasher hasher;
+
+  private long missedTrieKeys = 0;
+
+  public TrieKeyPreloader() {
+    this.hasher = new PedersenHasher();
+    this.trieKeyAdapter = new TrieKeyAdapter(hasher);
+    this.trieKeyAdapter.versionKey(
+        Address.ZERO); // TODO REMOVE is just to preload the native library for performance check
+  }
+
+  public HasherContext createPreloadedHasher(
+      final Address address,
+      final Map<StorageSlotKey, DiffBasedValue<UInt256>> storageUpdate,
+      final DiffBasedValue<Bytes> codeUpdate) {
+
+    // generate account triekeys
+    final List<Bytes32> accountKeyIds = new ArrayList<>(generateAccountKeyIds());
+
+    // generate storage triekeys
+    final List<Bytes32> storageKeyIds = new ArrayList<>();
+    boolean isStorageUpdateNeeded;
+    if (storageUpdate != null) {
+      final Set<StorageSlotKey> storageSlotKeys = storageUpdate.keySet();
+      isStorageUpdateNeeded = !storageSlotKeys.isEmpty();
+      if (isStorageUpdateNeeded) {
+        storageKeyIds.addAll(generateStorageKeyIds(storageSlotKeys));
+      }
+    }
+
+    // generate code triekeys
+    final List<Bytes32> codeChunkIds = new ArrayList<>();
+    boolean isCodeUpdateNeeded;
+    if (codeUpdate != null) {
+      final Bytes previousCode = codeUpdate.getPrior();
+      final Bytes updatedCode = codeUpdate.getUpdated();
+      isCodeUpdateNeeded =
+          !codeUpdate.isUnchanged() && !(codeIsEmpty(previousCode) && codeIsEmpty(updatedCode));
+      if (isCodeUpdateNeeded) {
+        accountKeyIds.add(Parameters.CODE_SIZE_LEAF_KEY);
+        codeChunkIds.addAll(
+            generateCodeChunkKeyIds(updatedCode == null ? previousCode : updatedCode));
+      }
+    }
+
+    return new HasherContext(
+        new CachedPedersenHasher(
+            generateManyTrieKeyHashes(address, accountKeyIds, storageKeyIds, codeChunkIds, true)),
+        !storageKeyIds.isEmpty(),
+        !codeChunkIds.isEmpty());
+  }
+
+  public void preLoadAccount(final Address account) {
+    CompletableFuture.runAsync(() -> cacheAccountTrieKeys(account));
+  }
+
+  public void preLoadStorageSlot(final Address account, final StorageSlotKey slotKey) {
+    CompletableFuture.runAsync(() -> cacheSlotTrieKeys(account, slotKey));
+  }
+
+  public void preLoadCode(final Address account, final Bytes code) {
+    CompletableFuture.runAsync(() -> cacheCodeTrieKeys(account, code));
+  }
+
+  public Map<Bytes32, Bytes32> generateManyTrieKeyHashes(
```

Comment:
do we really need to cache trieKeys?  seems like the only expensive thing to cache would be the 31 byte stem address, and we can just suffix it at-will.  It should reduce the cache size and might make for a better performing key generation on execution (since the stem would be cached and not regenerated if the slot key or code keys were not pre-calculated)